src/cpp/rtps/writer/LivelinessManager.cpp

```diff
@@ -61,6 +61,19 @@ bool LivelinessManager::add_writer(
         }
     }
     writers_.emplace_back(guid, kind, lease_duration);
+
+    if (!calculate_next())
+    {
+        timer_.cancel_timer();
+        return true;
+    }
+
+    // Some times the interval could be negative if a writer expired during the call to this function
+    // Once in this situation there is not much we can do but let asio timers expire inmediately
```

Comment:
It's misspelled in remove_writer as well
```suggestion
    // Once in this situation there is not much we can do but let asio timers expire immediately
```